src/tests/dbus-tests/test_80_filesystem.py

```diff
@@ -664,89 +678,130 @@ def test_mount_fstab_user(self):
         self.addCleanup(self.write_file, '/etc/fstab', fstab)
 
-        disk = self.get_object('/block_devices/' + os.path.basename(self.vdevs[0]))
-        self.assertIsNotNone(disk)
-
         # create filesystem
         disk.Format(self._fs_name, self.no_options, dbus_interface=self.iface_prefix + '.Block')
         self.addCleanup(self._clean_format, self.vdevs[0])
 
-        # create user for our test
-        self.addCleanup(self._remove_user, self.username)
-        uid, gid = self._add_user(self.username)
-
-        # add the disk to fstab
-        self._set_user_mountable(disk)
+        if fstab:
+            # add the disk to fstab
+            self._set_fstab_mountpoint(disk, options=fstab_options)
 
+    def _run_as_user_with_pipes(self, fn, uid, gid, device):
         # create pipe to get error (if any)
         parent_conn, child_conn = Pipe()
 
-        proc = Process(target=self._mount_as_user_fstab, args=(child_conn, int(uid), int(gid), self.vdevs[0]))
+        proc = Process(target=fn, args=(child_conn, int(uid), int(gid), device))
         proc.start()
         res = parent_conn.recv()
         parent_conn.close()
         proc.join()
 
+        return res
+
+    @unittest.skipUnless("JENKINS_HOME" in os.environ, "skipping test that modifies system configuration")
+    def test_mount_fstab_fail(self):
+        """ Test that user can't mount device not listed in /etc/fstab """
+        disk = self.get_object('/block_devices/' + os.path.basename(self.vdevs[0]))
+        self.assertIsNotNone(disk)
+
+        # create user for our test
+        self.addCleanup(self._remove_user, self.username)
+        uid, gid = self._add_user(self.username)
+
+        # add unmount cleanup now in case something wrong happens in the other process
+        self.addCleanup(self._unmount, self.vdevs[0])
+
+        # format the disk and add it to /etc/fstab
+        self._prepare_mount_test(disk, False, None)
+
+        # now try to mount the device as the user (should fail)
+        res = self._run_as_user_with_pipes(self._mount_as_user_fstab_fail, uid, gid, self.vdevs[0])
         if not res[0]:
             self.fail(res[1])
 
-    @unittest.skipUnless("JENKINS_HOME" in os.environ, "skipping test that modifies system configuration")
-    def test_mount_fstab_user_fail(self):
-        if not self._can_create:
-            self.skipTest('Cannot create %s filesystem' % self._fs_name)
+        # now mount it as root and test that user can't unmount it
+        mnt_path = disk.Mount(self.no_options, dbus_interface=self.iface_prefix + '.Filesystem')
+        self.assertIsNotNone(mnt_path)
+        self.assertTrue(os.path.ismount(mnt_path))
 
-        if not self._can_mount:
-            self.skipTest('Cannot mount %s filesystem' % self._fs_name)
+        # unmounting as the user should fail
+        res = self._run_as_user_with_pipes(self._unmount_as_user_fstab_fail, uid, gid, self.vdevs[0])
+        if not res[0]:
```

Comment:
`res` is expected to be `[False, "Some msg"]`, right? Shouldn't this condition be `if res[0]` then?

Reply:
No, because `_unmount_as_user_fstab_fail` will fail when it fails to fail to unmount the device.